### DISCUSSION TITLE
Difference between HTTPException and Response

### DISCUSSION THREAD

**Discussion by riwu** [encode/starlette#2567] [Q&A] [answered]
Is there any difference between `return Response(status_code=404)` and `raise HTTPException(status_code=404)`?  
Which is recommended and why?

**Comment by hasansezertasan** [comment, marked as answer]
Simply, the response with status code would return directly, raised errors would get caught by error handlers.

I'd go with returning a custom response that describes what went wrong instead of raising an error.

Or I would write an exception handler that caughts the exception and returns a custom response that describes what went wrong.

I believe it's just a design decision you need to make.
